test/fixtures/wpt/encoding/encodeInto.any.js

```diff
@@ -0,0 +1,149 @@
+      "destinationOffset": 4,
+      "filler": 0
+    },
+    {
+      "bufferIncrease": 0,
+      "destinationOffset": 0,
+      "filler": 0x80
+    },
+    {
+      "bufferIncrease": 10,
+      "destinationOffset": 4,
+      "filler": 0x80
+    },
+    {
+      "bufferIncrease": 0,
+      "destinationOffset": 0,
+      "filler": "random"
+    },
+    {
+      "bufferIncrease": 10,
+      "destinationOffset": 4,
+      "filler": "random"
+    }
+  ].forEach(destinationData => {
+    test(() => {
+      // Setup
+      const bufferLength = testData.destinationLength + destinationData.bufferIncrease,
+            destinationOffset = destinationData.destinationOffset,
+            destinationLength = testData.destinationLength,
+            destinationFiller = destinationData.filler,
+            encoder = new TextEncoder(),
+            buffer = new ArrayBuffer(bufferLength),
+            view = new Uint8Array(buffer, destinationOffset, destinationLength),
+            fullView = new Uint8Array(buffer),
+            control = new Array(bufferLength);
+      let byte = destinationFiller;
+      for (let i = 0; i < bufferLength; i++) {
+        if (destinationFiller === "random") {
+          byte = Math.floor(Math.random() * 256);
+        }
+        control[i] = byte;
+        fullView[i] = byte;
+      }
+
+      // It's happening
+      const result = encoder.encodeInto(testData.input, view);
+
+      // Basics
+      assert_equals(view.byteLength, destinationLength);
+      assert_equals(view.length, destinationLength);
+
+      // Remainder
+      assert_equals(result.read, testData.read);
+      assert_equals(result.written, testData.written.length);
+      for (let i = 0; i < bufferLength; i++) {
+        if (i < destinationOffset || i >= (destinationOffset + testData.written.length)) {
+          assert_equals(fullView[i], control[i]);
+        } else {
+          assert_equals(fullView[i], testData.written[i - destinationOffset]);
+        }
+      }
+    }, "encodeInto() with " + testData.input + " and destination length " + testData.destinationLength + ", offset " + destinationData.destinationOffset + ", filler " + destinationData.filler);
+  });
+});
+
+[DataView,
+ Int8Array,
+ Int16Array,
+ Int32Array,
+ Uint16Array,
+ Uint32Array,
+ Uint8ClampedArray,
+ Float32Array,
+ Float64Array].forEach(view => {
+  test(() => {
+    assert_throws(new TypeError(), () => new TextEncoder().encodeInto("", new view(new ArrayBuffer(0))));
+  }, "Invalid encodeInto() destination: " + view);
+ });
+
+test(() => {
+  assert_throws(new TypeError(), () => new TextEncoder().encodeInto("", new ArrayBuffer(10)));
+}, "Invalid encodeInto() destination: ArrayBuffer");
+
+test(() => {
+  const buffer = new ArrayBuffer(10),
+        view = new Uint8Array(buffer);
+  let { read, written } = new TextEncoder().encodeInto("", view);
+  assert_equals(read, 0);
+  assert_equals(written, 0);
+  new MessageChannel().port1.postMessage(buffer, [buffer]);
```

Comment:
I try to finish the  feature encodeInto, after I finished I tested it with the test cases, but meet the failure as followed

```
[FAILURE] encodeInto() and a detached output buffer
MessageChannel is not defined
ReferenceError: MessageChannel is not defined
at Test. (node/test/fixtures/wpt/encoding/encodeInto.any.js:142:3) 
```